src/btree/bt_cursor.c

```diff
@@ -1019,6 +1019,13 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
             __cursor_state_restore(cursor, &state);
         else {
             __wt_value_return(cbt, cbt->upd_value);
+            if (WT_CURSOR_BOUNDS_SET(cursor)) {
+                if (btree->type == BTREE_ROW)
+                    WT_ERR(
+                      __wt_compare(session, btree->collator, &cursor->key, &state.key, &exact));
```

Comment:
I think this should be comparing against a bound not state.key which was a prefix search mechanism, I could be mistaken though.

Reply:
Oh right because the search key set is state.key and we're only interested in knowing what side we're on.

Reply:
Can you add a comment explaining why we need this extra compare in the bounded cursor case?

Reply:
[5b00b50](https://github.com/wiredtiger/wiredtiger/pull/8231/commits/5b00b50496e7259fd3ca6808621d0aa24595aadf)